src/Sarif.Sarifer/SpamBackgroundAnalyzer.cs

```diff
@@ -132,5 +133,25 @@ protected override void AnalyzeCore(Uri uri, string text, string solutionDirecto
                 sarifLogger.Log(rule, result);
             }
         }
+
+        private void LoadFiles()
```

Comment:
>private [](start = 8, length = 7)

This is unit testable. Make it `internal` (with a comment explaining why).

New project: `Sarif.Sarifer.UnitTests`

Use mock file system to prepare the content you read from the solution directory.

Reply:
To make this more testable:

1. `LoadFiles` is `static` so you don't have to instantiate the object.
2. You pass the directory as a parameter.
3. You return the list of rules.
4. You inject the file system into this method.

---
In reply to: [531805897](https://github.com/microsoft/sarif-visualstudio-extension/pull/290#discussion_r531805897) [](ancestors = 531805897)

Reply:
Just added the tests

---
In reply to: [531806339](https://github.com/microsoft/sarif-visualstudio-extension/pull/290#discussion_r531806339) [](ancestors = 531806339,531805897)